fastdup/fastdup_controller.py

```diff
@@ -1260,6 +1260,26 @@ def _verify_fastdup_run_args(self, input_dir, work_dir, df_annot, subset, data_t
             else:
                 assert False, f"Wrong data type {data_type}"
 
+    def caption(self, modelname='automatic', subset: list = None, kwargs=None) -> pd.DataFrame:
+        if not self._fastdup_applied:
+            raise RuntimeError('Fastdup was not applied yet, call run() first')
+
+        df = self.annotations(valid_only=True)
+        assert len(df), "No images found."
+
+        if modelname in FD.CAPTION_MODEL_NAMES:
+            from fastdup.captions import generate_labels
+            df['caption'] = generate_labels(df['filename'], modelname)
+        elif modelname == FD.VQA_MODEL1_NAME:
+            from fastdup.captions import generate_vqa_labels
+            df['caption'] = generate_vqa_labels(df['filename'], "Is the photo taken indoors or outdoors", kwargs)
```

Comment:
Let users specify prompts?

Reply:
thanks, I have added functionality to let the user specify the prompt, and added an example of this to the notebook